pkg/resources/object_renaming_lists_and_sets_acceptance_test.go

```diff
@@ -0,0 +1,463 @@
+						{"name": "column3", "string": "222", "int": "222"},
+						{"name": "column1", "string": "111", "int": "111"},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_ListsWithDuplicatedItems(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableObjectRenamingTest)
+	acc.TestAccPreCheck(t)
+
+	// Fails, because the SuppressDiffFunc works on the hash of individual items.
+	// To correctly suppress such changes, the number of repeated hashes should be counted.
+	t.Skip("Currently failing, because duplicated hashes are not supported.")
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: objectRenamingConfigList([]map[string]any{
+					{"string": "111", "int": 111},
+					{"string": "222", "int": 222},
+					{"string": "333", "int": 333},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+						{"string": "111", "int": "111"},
+						{"string": "222", "int": "222"},
+						{"string": "333", "int": "333"},
+					}),
+				),
+			},
+			// Introduce duplicates
+			{
+				Config: objectRenamingConfigList([]map[string]any{
+					{"string": "111", "int": 111},
+					{"string": "111", "int": 111},
+					{"string": "222", "int": 222},
+					{"string": "222", "int": 222},
+					{"string": "333", "int": 333},
+					{"string": "333", "int": 333},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+						{"string": "111", "int": "111"},
+						{"string": "111", "int": "111"},
+						{"string": "222", "int": "222"},
+						{"string": "222", "int": "222"},
+						{"string": "333", "int": "333"},
+						{"string": "333", "int": "333"},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func objectRenamingConfigList(listItems []map[string]any) string {
+	generateListItem := func(name *string, s string, i int) string {
+		var nameField string
+		if name != nil {
+			nameField = fmt.Sprintf(`name = "%s"`, *name)
+		}
+		return fmt.Sprintf(`
+list {
+	%[1]s
+	string = "%[2]s"
+	int = %[3]d
+}
+`, nameField, s, i)
+	}
+
+	generatedListItems := ""
+	for _, item := range listItems {
+		var name *string
+		if nameValue, ok := item["name"]; ok && nameValue != nil {
+			name = sdk.String(nameValue.(string))
+		}
+		generatedListItems += generateListItem(name, item["string"].(string), item["int"].(int))
+	}
+
+	return fmt.Sprintf(`
+resource "snowflake_object_renaming" "test" {
+	%s
+}
+`, generatedListItems)
+}
+
+func TestAcc_BasicManuallyOrderedListFlow(t *testing.T) {
```

Comment:
it would be nice to see this in action also for item with name + type (like e.g. argument/column) + order so that we check how much work is needed to handle any number of attributes in the given item.

Reply:
As discussed, added type and tested all the supported and unsupported cases we talked about.